File: .scratch-theme-test/sample.ts
import { readFileSync } from 'fs';
import { join } from 'path';

// A constant config object
const CONFIG = {
	retries: 3,
	timeout: 5000,
	label: 'theme-test',
};

export type Status = 'idle' | 'running' | 'done';

export interface Task {
	id: number;
	name: string;
	status: Status;
}

export function loadTasks(dir: string): Task[] {
	const raw = readFileSync(join(dir, 'tasks.json'), 'utf8');
	const parsed = JSON.parse(raw) as Task[];
	return parsed.filter(t => t.status !== 'done');
}

class Scheduler {
	private tasks: Task[] = [];

	add(task: Task): void {
		this.tasks.push(task);
	}

	run(): number {
		let count = 0;
		for (const task of this.tasks) {
			console.log(`running ${task.name}`);
			count++;
		}
		return count;
	}
}

const scheduler = new Scheduler();
scheduler.add({ id: 1, name: 'build', status: 'idle' });

File: .scratch-theme-test/sample.cpp
#include <iostream>
#include <vector>
#include <string>

namespace theme {

struct Task {
	int id;
	std::string name;
	bool done = false;
};

class Scheduler {
public:
	void add(const Task& task) {
		tasks_.push_back(task);
	}

	int run() const {
		int count = 0;
		for (const auto& task : tasks_) {
			std::cout << "running " << task.name << '\n';
			++count;
		}
		return count;
	}

private:
	std::vector<Task> tasks_;
};

}  // namespace theme

int main() {
	theme::Scheduler scheduler;
	scheduler.add({1, "build", false});
	return scheduler.run();
}
